### PR TITLE
[JUJU-2232] Remove charm store support from facades

### PR DESCRIPTION
**Pull request by hmlanigan**
Finish removing charmstore support from the facades. Most work is to remove specific cases for charm store and update the unit tests appropriately.

Follow up will include removing the macaroons.

Continues the work of #14947 and #14948.

## QA steps

No change to current behavior.

### CODE REVIEW

File: apiserver/facades/agent/caasoperator/mock_test.go [hunk 1]
```diff
@@ -44,7 +44,7 @@ func newMockState() *mockState {
 			name: "gitlab",
 			life: state.Alive,
 			charm: mockCharm{
-				url:    charm.MustParseURL("cs:gitlab-1"),
+				url:    charm.MustParseURL("ch:gitlab-1"),
```
Comment (cderici) [inline apiserver/facades/agent/caasoperator/mock_test.go:47]: we don't have revisions in the url anymore, right?

**Review by cderici — APPROVED** [review]
LGTM

File: apiserver/facades/agent/caasoperator/mock_test.go (hunk 1 above)
Comment (hmlanigan) [reply apiserver/facades/agent/caasoperator/mock_test.go:47]: We do not allow users to specify ch:<charm>-<revision> to deploy a charm, either on the CLI or inside of a bundle. However we do use it internally to juju.

**Pull request merged**